tests/src/test/java/com/maxio/advancedbilling/controllers/subscriptiongroups/SubscriptionGroupsControllerFindTest.java

```diff
@@ -0,0 +1,135 @@
+import com.maxio.advancedbilling.models.AccountBalance;
+import com.maxio.advancedbilling.models.Component;
+import com.maxio.advancedbilling.models.Customer;
+import com.maxio.advancedbilling.models.FullSubscriptionGroupResponse;
+import com.maxio.advancedbilling.models.Product;
+import com.maxio.advancedbilling.models.ProductFamily;
+import com.maxio.advancedbilling.models.Subscription;
+import com.maxio.advancedbilling.models.SubscriptionGroupBalances;
+import com.maxio.advancedbilling.models.SubscriptionGroupCustomer;
+import com.maxio.advancedbilling.models.SubscriptionGroupSignupResponse;
+import com.maxio.advancedbilling.utils.TestSetup;
+import com.maxio.advancedbilling.utils.TestTeardown;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.maxio.advancedbilling.utils.assertions.CommonAssertions.assertNotFound;
+import static com.maxio.advancedbilling.utils.assertions.CommonAssertions.assertUnauthorized;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SubscriptionGroupsControllerFindTest {
+
+    private static final TestSetup TEST_SETUP = new TestSetup();
+    private static final AdvancedBillingClient CLIENT = TestClient.createClient();
+    private static final SubscriptionGroupsController SUBSCRIPTION_GROUPS_CONTROLLER = CLIENT.getSubscriptionGroupsController();
+    private static SubscriptionGroupSignupResponse groupSignup;
+    private static Product product;
+    private static Component meteredComponent;
+    private static Customer customer;
+
+    @BeforeAll
+    static void setUp() throws IOException, ApiException {
+        ProductFamily productFamily = TEST_SETUP.createProductFamily();
+        product = TEST_SETUP.createProduct(productFamily, b -> b.priceInCents(1250));
+        meteredComponent = TEST_SETUP.createMeteredComponent(productFamily, 11.5);
+        customer = new TestSetup().createCustomer();
+    }
+
+    @AfterAll
+    static void deleteCustomer() throws IOException, ApiException {
+        new TestTeardown().deleteCustomer(customer);
+    }
+
+    @AfterEach
+    void cleanupSubscriptionGroup() throws IOException, ApiException {
+        if (groupSignup != null) {
+            new TestTeardown().deleteSubscriptionGroup(groupSignup);
+            groupSignup = null;
+        }
+    }
+
+    @Test
+    void shouldFindSubscriptionGroup() throws IOException, ApiException {
+        // given
+        groupSignup = new TestSetup().signupWithSubscriptionGroup(product, meteredComponent);
+        Integer primarySubscriptionId = groupSignup.getPrimarySubscriptionId();
+        List<Integer> secondarySubscriptionIds = new ArrayList<>(groupSignup.getSubscriptionIds());
+        Integer secondarySubscriptionId = secondarySubscriptionIds.get(0);
```

Comment:
in other test you've used this helper to extract secondary sub id
```
private Integer getSecondarySubscriptionId(SubscriptionGroupSignupResponse signupResponse) {
        ArrayList<Integer> subscriptionIds = new ArrayList<>(signupResponse.getSubscriptionIds());
        subscriptionIds.remove(signupResponse.getPrimarySubscriptionId());
        return subscriptionIds.get(0);
    }
```
here these 2 can be the same subscription 
<img width="423" alt="image" src="https://github.com/user-attachments/assets/a3e984c3-ce08-472f-84ce-a7b2bddc9548">